drivers/lte_link_control/lte_lc.c

```diff
@@ -63,107 +64,66 @@ static const char cgdcont[] = "AT+CGDCONT="CONFIG_LTE_PDP_CONTEXT;
 static const char legacy_pco[] = "AT%XEPCO=0";
 #endif
 
-static int at_cmd(int fd, const char *cmd, size_t size)
+void at_handler(char *response)
 {
-	int len;
-	u8_t buffer[LC_MAX_READ_LENGTH];
+	LOG_DBG("recv: %s", response);
 
-	LOG_DBG("send: %s", cmd);
-	len = send(fd, cmd, size, 0);
-	if (len != size) {
-		LOG_ERR("send: failed");
-		return -EIO;
-	}
-
-	len = recv(fd, buffer, LC_MAX_READ_LENGTH, 0);
-	if ((len < AT_CMD_SIZE(success)) ||
-	    (memcmp(success, buffer, AT_CMD_SIZE(success)) != 0)) {
-		LOG_ERR("recv: %s", buffer);
-		return -EIO;
+	if (!memcmp(status1, response, AT_CMD_SIZE(status1)) ||
```

Comment:
What about using the AT parser here instead?

Reply:
Sure, but I am not sure if that is within the scope of this PR. I just ported the code to use the common at_cmd driver instead of the self hosted AT command driver. That is maybe something that the AT parser PR can cover or a separate improvement PR.

Reply:
That is something we can improve later.